src/apps/service/apps/jobs/submissions.clj

```diff
@@ -179,65 +117,31 @@
-       (into {})))
-
-(defn- format-submission-in-batch
-  [submission job-number path-map]
-  (let [job-suffix (str "analysis-" (inc job-number))]
-    (assoc (update-in submission [:config] (partial substitute-param-values path-map))
-      :group      (config/jex-batch-group-name)
-      :name       (str (:name submission) "-" job-suffix)
-      :output_dir (ft/path-join (:output_dir submission) job-suffix))))
-
-(defn- submit-job-in-batch
-  [apps-client user submission paths-exist job-number path-map]
-  (when (every? (partial get paths-exist) (map keyword (vals path-map)))
-    (let [job-info (.submitJob apps-client (format-submission-in-batch submission job-number path-map))]
-      (perms-client/register-private-analysis (:shortUsername user) (:id job-info))
-      job-info)))
-
-(defn- preprocess-batch-submission
-  [submission output-dir parent-id]
-  (assoc submission
-    :output_dir           output-dir
-    :parent_id            parent-id
-    :create_output_subdir false))
+  (transaction
+    (let [batch-id (save-batch* user app submission output-dir)]
+      (save-batch-step batch-id (first job-types))
+      (perms-client/register-private-analysis (:shortUsername user) batch-id)
+      batch-id)))
+
+(defn- pre-submit-batch-validation
+  [input-params-by-id input-paths-by-id path-list-stats]
+  (doseq [path-stat path-list-stats]
+    (validate-path-list-stats path-stat))
+
+  (->> (extract-ht-param-ids path-list-stats input-paths-by-id)
+       (select-keys input-params-by-id)
+       vals
+       validate-ht-params))
 
 (defn- submit-batch-job
   [apps-client user input-params-by-id input-paths-by-id path-list-stats job-types app submission]
-  (dorun (map validate-path-list-stats path-list-stats))
-  (let [ht-param-ids (extract-ht-param-ids path-list-stats input-paths-by-id)
-        _            (validate-ht-params (vals (select-keys input-params-by-id ht-param-ids)))
-        ht-paths     (set (map :path path-list-stats))
-        path-lists   (get-path-list-contents-map user ht-paths)
-        paths-exist  (validate-path-lists user path-lists)
-        path-maps    (map-slices path-lists)
-        output-dir   (get-batch-output-dir user submission)
-        batch-id     (save-batch user job-types app submission output-dir)
-        submission   (preprocess-batch-submission submission output-dir batch-id)]
-    (dorun (map-indexed (partial submit-job-in-batch apps-client user submission paths-exist) path-maps))
-    (-> (job-listings/list-job apps-client batch-id)
-        (assoc :missing-paths (extract-missing-paths paths-exist))
-        remove-nil-values)))
+  (pre-submit-batch-validation input-params-by-id input-paths-by-id path-list-stats)
+
+  (let [ht-paths   (set (map :path path-list-stats))
+        output-dir (get-batch-output-dir user submission)
+        batch-id   (save-batch user job-types app submission output-dir)]
+    (async/submit-batch-jobs apps-client user ht-paths submission output-dir batch-id)
```

Comment:
This is the call that submits the HT Path List and sub-job processing to a `future` call, but we ignore the result of that `future` so we don't block and this function can respond to the client right away.